CalendarApp/Models/Event.swift

```diff
@@ -9,8 +9,8 @@ import Foundation
 import CalendarKit
 
 @objcMembers
-public class Event : NSObject {
-    public var parseObjectId: String?
+public class Event : NSObject, NSSecureCoding {
```

Comment:
I think I would have made an intermediate representation to avoid having to add all this to the canonical model. This can be almost considered a type of remote. You have your canonical model, that can be built from all sorts of other representations including a securely decoded representation.